### PR TITLE
Automatic releasing

### PR DESCRIPTION
**Pull request by martinRenou**
cc. @jtpio @ianhi

### CODE REVIEW

File: dev-environment.yml [hunk 1]
```diff
@@ -8,3 +8,4 @@ dependencies:
     - jupyterlab=3
     - ipywidgets>=7.6
     - flake8
```
Comment (ianhi) [inline dev-environment.yml:10]: maybe explicitly need `wheel`?

Comment (martinRenou) [reply dev-environment.yml:10]: Thanks for the review!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        channels: conda-forge
+
+    - name: Build package
+      run: python setup.py sdist bdist_wheel
+
+    - name: Publish the Python package
+      env:
+        TWINE_USERNAME: __token__
+        TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
+      run: twine upload dist/ipycanvas*
```
Comment (jtpio) [inline .github/workflows/release.yml:35]: There is also an action for this part: https://github.com/pypa/gh-action-pypi-publish

Although using `twine` directly also sounds good.

Comment (martinRenou) [reply .github/workflows/release.yml:35]: Thanks! I am a bit worried the action would publish other things from the `dist` folder and fail.

In the widget-ts-cookiecutter the `dist` folder contains JS dist code and the wheels. We might want to fix it at some point.

Comment (jtpio) [reply .github/workflows/release.yml:35]: Good point :+1: 

Using the action doesn't save that much code and effort anyway it seems.

**Comment by ianhi** [comment]
Once this is tested here perhaps it should be added to the cookiecutter? (and ipympl of course :smile: )

**Comment by martinRenou** [comment]
> Once this is tested here perhaps it should be added to the cookiecutter? (and ipympl of course smile )

Yep!!

**Comment by martinRenou** [comment]
Ok. Let's merge, release, and hope for the best.

**Comment by jtpio** [comment]
> Once this is tested here perhaps it should be added to the cookiecutter? (and ipympl of course smile )

Yes, and there was some discussions about this in https://github.com/jupyterlab/extension-cookiecutter-ts/issues/101 too.

**Pull request merged**